From e490a42ab469fd325b9afb7f6211f4cf20b7c9f8 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 2 Apr 2015 11:39:20 +0200
Subject: [PATCH 02/42] base: make MSVC happy with __PRETTY_FUNCTION__

We will be printing some few debug lines to stdout/stderr for
_developer_ benefit.
---
 base/logging.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/base/logging.h b/base/logging.h
index aa243a9..9dae516 100644
--- a/base/logging.h
+++ b/base/logging.h
@@ -912,6 +912,11 @@ inline std::ostream& operator<<(std::ostream& out, const std::wstring& wstr) {
 #define NOTIMPLEMENTED_MSG "NOT IMPLEMENTED"
 #endif
 
+#ifdef _MSC_VER
+#	define __PRETTY_FUNCTION__ __FUNCSIG__
+#	define __func__ __FUNCSIG__
+#endif
+
 #if NOTIMPLEMENTED_POLICY == 0
 #define NOTIMPLEMENTED() EAT_STREAM_PARAMETERS
 #elif NOTIMPLEMENTED_POLICY == 1
-- 
2.4.0

